releasenotes/notes/deprecate-classical-function-3bf44ef26d984366.yaml

```diff
@@ -0,0 +1,25 @@
+---
+deprecations_circuits:
+  - |
+    The :mod:`qiskit.circuit.classicalfunction` and with it
+    the :class:`.ClassicalFunction`, its related :func:`.classical_function`
+    and :class:`.BooleanExpression` have been deprecated as of Qiskit 1.4 
+    and will be removed in Qiskit 2.0. This change is done to avoid a 
+    dependency on the external library `tweedledum`, which is no longer 
+    compatible with all of Qiskit's supported platforms and Python versions.
+    For a similar functionality please use the :class:`.PhaseOracle` which
+    is going to have a `tweedledum` independent implementation, and
+    the :class:`.BitFlipOracle` which will be added in Qiskit 2.0.
+
+    Until :class:`.BitFlipOracle` is added, a phase flip oracle can be converted
+    to a bit flip oracle by conditioning it on the result qubit, and applying 
+    Hadamard gate before and after the application of the oracle, as in the
+    following example (where the oracle is on `qr_x` and the result is on `qr_y`):
+
+    .. code-block:: python
+    
+        phase_flip_oracle = PhaseOracle(bool_expr)
```

Comment:
That's nice! What do you think of making this a fully executable snippet with some example boolean expression?

Reply:
Done in b4878c62c